src/frontmatter/frontmatter.ts

```diff
@@ -30,7 +38,8 @@ export class Frontmatter<T extends object = any> implements IFrontmatter<T> {
     this.document = root;
     if (root.children.length > 0 && root.children[0].type === 'yaml') {
       this.node = root.children[0] as Unist.Literal;
-      this.properties = safeParse(this.node.value as string);
+      // typings are a bit tricked, but let's move the burden of validation to consumer
+      this.properties = safeParse<Partial<T>>(this.node.value as string);
```

Comment:
and nullify properties and handle the nullified properties in `getAll`, etc.
I believe we started to return empty object to make it easier to integrate it in Studio, the more significant frontmatter consumer?
Don't remember exactly.